apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_producer.erl

```diff
@@ -74,36 +79,8 @@ on_start(InstId, Config) ->
         ssl => C(ssl)
     },
     ClientId = InstId,
-    ok = emqx_resource:allocate_resource(InstId, ?kafka_client_id, ClientId),
-    case wolff:ensure_supervised_client(ClientId, Hosts, ClientConfig) of
-        {ok, _} ->
-            case wolff_client_sup:find_client(ClientId) of
-                {ok, Pid} ->
-                    case wolff_client:check_connectivity(Pid) of
-                        ok ->
-                            ok;
-                        {error, Error} ->
-                            deallocate_client(ClientId),
-                            throw({failed_to_connect, Error})
-                    end;
-                {error, Reason} ->
-                    deallocate_client(ClientId),
-                    throw({failed_to_find_created_client, Reason})
-            end,
-            ?SLOG(info, #{
-                msg => "kafka_client_started",
-                instance_id => InstId,
-                kafka_hosts => Hosts
-            });
-        {error, Reason} ->
-            ?SLOG(error, #{
-                msg => failed_to_start_kafka_client,
-                instance_id => InstId,
-                kafka_hosts => Hosts,
-                reason => Reason
-            }),
-            throw(failed_to_start_kafka_client)
-    end,
+    ok = ensure_client(ClientId, Hosts, ClientConfig),
+    emqx_resource:allocate_resource(InstId, ?kafka_client_id, ClientId),
```

Comment:
Nit: I think we might allocate the client_id first, so that if the client creation fails midway, we still have the "handle" to destroy it later.

```suggestion
    emqx_resource:allocate_resource(InstId, ?kafka_client_id, ClientId),
    ok = ensure_client(ClientId, Hosts, ClientConfig),
```